commit c023cc9edaf0aaece9f5fc879fd303f148536c31
Author:     Yuan-Ting Hsieh (謝沅廷) <yuantinghsieh@gmail.com>
AuthorDate: Thu Apr 21 13:36:44 2022 -0700
Commit:     GitHub <noreply@github.com>
CommitDate: Thu Apr 21 13:36:44 2022 -0700

    Add default env var PYTHONPATH if it is not set (#426)
---
 nvflare/private/fed/client/client_executor.py | 2 +-
 nvflare/private/fed/server/server_engine.py   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/client/client_executor.py b/nvflare/private/fed/client/client_executor.py
index 105f335e..c8aab004 100644
--- a/nvflare/private/fed/client/client_executor.py
+++ b/nvflare/private/fed/client/client_executor.py
@@ -191,7 +191,7 @@ class ProcessExecutor(ClientExecutor):
 
         new_env = os.environ.copy()
         if app_custom_folder != "":
-            new_env["PYTHONPATH"] = new_env["PYTHONPATH"] + ":" + app_custom_folder
+            new_env["PYTHONPATH"] = new_env.get("PYTHONPATH", "") + os.pathsep + app_custom_folder
 
         command_options = ""
         for t in args.set:
diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index e2148529..019abafd 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -248,7 +248,7 @@ class ServerEngine(ServerEngineInternalSpec):
     ):
         new_env = os.environ.copy()
         if app_custom_folder != "":
-            new_env["PYTHONPATH"] = new_env["PYTHONPATH"] + ":" + app_custom_folder
+            new_env["PYTHONPATH"] = new_env.get("PYTHONPATH", "") + os.pathsep + app_custom_folder
 
         listen_port = open_ports[1]
         if snapshot:
